tests/vmi_multus_test.go

```diff
@@ -1008,16 +1042,75 @@ var _ = Describe("[Serial]Macvtap", func() {
 
 		BeforeEach(func() {
 			macAddress := "02:03:04:05:06:07"
-			macvtapVMI = createCirrosVMIWithMacvtapDefinedMAC(virtClient, macvtapNetworkName, macAddress)
+			clientVMI = createCirrosVMIStaticIPRandomNode(virtClient, macvtapNetworkName, macAddress)
 		})
 
 		It("should be successful when the VMI MAC address is defined in its spec", func() {
 			By("starting the migration")
-			migration := tests.NewRandomMigration(macvtapVMI.GetName(), macvtapVMI.GetNamespace())
+			migration := tests.NewRandomMigration(clientVMI.GetName(), clientVMI.GetNamespace())
 			migrationUID := tests.RunMigrationAndExpectCompletion(virtClient, migration, migrationWaitTime)
 
 			// check VMI, confirm migration state
-			tests.ConfirmVMIPostMigration(virtClient, macvtapVMI, migrationUID)
+			tests.ConfirmVMIPostMigration(virtClient, clientVMI, migrationUID)
+		})
+
+		Context("with live traffic", func() {
+			var serverVMI *v1.VirtualMachineInstance
+			var serverIP string
+
+			getVMMacvtapIfaceIP := func(vmName string, namespace string, macAddress string) (string, error) {
+				var vmiIP string
+				err := wait.PollImmediate(time.Second, 2*time.Minute, func() (done bool, err error) {
+					vmi, err := virtClient.VirtualMachineInstance(namespace).Get(vmName, &v13.GetOptions{})
+					if err != nil {
+						return false, err
+					}
+
+					for _, iface := range vmi.Status.Interfaces {
+						if iface.MAC == macAddress {
+							vmiIP = iface.IP
+							return true, nil
+						}
+					}
+
+					return false, nil
+				})
+				if err != nil {
+					return "", err
+				}
+
+				// code below should be removed if / when KubeVirt honors its
+				// API and returns an IP on the Interface.IP attribute
+				// Track the following PR to know when to update:
+				//   - https://github.com/kubevirt/kubevirt/pull/4217
+				ipAndNetworkLength := strings.Split(vmiIP, "/")
+				if len(ipAndNetworkLength) != 2 {
+					return "", fmt.Errorf("could not compute the IP address from the CIDR")
+				}
+				return ipAndNetworkLength[0], err
+			}
+
+			BeforeEach(func() {
+				macAddress := "02:03:04:05:06:aa"
+
+				serverVMI = createFedoraVMIRandomNode(virtClient, macvtapNetworkName, macAddress)
+				Expect(serverVMI.Status.Interfaces).NotTo(BeEmpty(), "a migrate-able VMI must have network interfaces")
+
+				serverIP, err = getVMMacvtapIfaceIP(serverVMI.GetName(), serverVMI.GetNamespace(), macAddress)
```

Comment:
Not sure what you are refering to, I was talking about `serverVMI.GetName()` and `serverVMI.GetNamespace()` that can just be `serverVMI.Name` and `serverVMI.Namespace`.

Reply:
I'd rather used a method on the public API rather than directly access an attribute. 

It's being done in plenty of places, including out best practices reference test in https://github.com/kubevirt/kubevirt/blob/6abcfd5595fb716224e9eb1220eb2ef0501d0fc6/tests/network/primary_pod_network.go#L150 .

Never the less, I think we can agree that both work.